src/CMakeLists.txt

```diff
@@ -12,4 +12,29 @@ project(vocoder VERSION 0.0)
 set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_CXX_STANDARD_REQUIRED True)
 
-add_executable(vocoder src/main.cpp)
\ No newline at end of file
+include(FetchContent)
+FetchContent_Declare(
+	googletest
+	URL https://github.com/google/googletest/archive/03597a01ee50ed33e9dfd640b249b4be3799d395.zip
+	)
+
+# For Windows: Prevent overriding the parent project's compiler/linker settings
+set(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
+FetchContent_MakeAvailable(googletest)
+
+add_executable(vocoder main.cpp)
+
+enable_testing()
+
+add_executable(
+    example_test
+    unit_tests/example_test.cc
+    )
+target_link_libraries(
+    example_test
+    GTest::gtest_main
+    )	

```

Comment:
Do we need to do this for every test? I was under the impression gtest handled test finding automatically

Reply:
I think there might be a way to point gtest to the unit test directory instead, but I believe each test  has to be linked manually in CMake.

Reply:
It sounds like I assumed too much of gtest. Apparently the impressive part is that it finds and runs all the tests in the binary, you still need to link the binary.